### PR TITLE
Refactor to consolidate api methods

### PR DESCRIPTION
**Pull request by bf4**
Extracted from #67

### CODE REVIEW

File: lib/qbo_api.rb [hunk 1]
```diff
@@ -62,30 +56,6 @@ def connection(url: get_endpoint)
...
-
-  def deactivate(entity, id:)
-    err_msg = "Deactivate is only for name list entities. Use .delete instead"
-    raise QboApi::NotImplementedError.new, err_msg unless is_name_list_entity?(entity)
-    payload = set_deactivate(entity, id)
-    request(:post, entity: entity, path: entity_path(entity), payload: payload)
-  end
-
   # https://developer.intuit.com/docs/0100_quickbooks_online/0100_essentials/0085_develop_quickbooks_apps/0004_authentication_and_authorization/oauth_management_api#/Reconnect
   def disconnect
```
Comment (bf4) [inline lib/qbo_api.rb]: I don't know where these disconnect/reconnect methods should go. Are they part of the legacy oauth api or some 'app' api?

Comment (ntippie) [reply lib/qbo_api.rb]: `#reconnect` is part of the Oauth1 API (which we need to support through the end of 2018), and `#disconnect` is the equivalent of "revoke" in Oauth2. I think they should go in with the connection code from #70.

**Comment by ntippie** [comment]
I think we should rename the `params` argument when used for queries. There are other places in the gem where `params` is used to add URL params to the connection request, like if you need to add `?include=void`.

**Comment by minimul** [comment]
re: params. Right.

File: lib/qbo_api.rb (hunk 1 above)
Comment (bf4) [reply lib/qbo_api.rb]: so, they should go in the connection methods, maybe?

Comment (bf4) [reply lib/qbo_api.rb]: oh, oops, you said that

**Comment by bf4** [comment]
@ntippie it could be renamed to `query_param`, but that would be a breaking change

**Comment by ntippie** [comment]
Sorry, I think I misread your code. Your argument for those query params I was referring to is `type`. No conflict here. Is there a better argument name than `type` that applies well? Maybe `criteria`?

**Comment by bf4** [comment]
> Is there a better argument name than `type` that applies well? Maybe `criteria`?

@ntippie Oh, that's pre-existing. I agree that it's not very descriptive. It's more like `id_or_filter_args`

**Comment by minimul** [comment]
I like `id_or_filter_args`.

**Comment by bf4** [comment]
@minimul Added ac93dcf65efb859ed7ee3ec85309bb4e8796ac89 per feedback and rebased off of master

**Comment by minimul** [comment]
Can you rebase with master again.

**Comment by bf4** [comment]
rebased and pushed

**Pull request merged**